### PR TITLE
Default storage namespace

### PR DESCRIPTION
**Pull request by johnnyaug**
Added a new configuration `blockstore.default_namespace_prefix`.
When specified, the storage namespace field in the UI will be auto-filled.
The specified value will be the prefix, and the repository name will be added to it.

image

### Extra
* Added a (?) icon near the storage namespace field, linking to the docs.
* Small docs improvements.

### CODE REVIEW

File: docs/reference/configuration.md
Comment (arielshaqed) [inline docs/reference/configuration.md:56]: "If missing, ..."?

File: docs/reference/object-model.md
Comment (arielshaqed) [inline docs/reference/object-model.md:149]: I prefer the original statement.

File: webui/src/lib/components/repositoryCreateForm.jsx
Comment (arielshaqed) [inline webui/src/lib/components/repositoryCreateForm.jsx:24]: Not sure the prefix itself is necessarily "valid"; don't we still need to check?

**Review by talSofer — COMMENTED** [review]
Very nice!

> The specified value will be the prefix, and the repository name will be added to it.

Why are we adding the repository name to the prefix? to create a separate storage space for each repository (if so, why)?

File: webui/src/lib/components/repositoryCreateForm.jsx
Comment (arielshaqed) [inline webui/src/lib/components/repositoryCreateForm.jsx:63]: ?

File: webui/src/lib/components/repositoryCreateForm.jsx
Comment (arielshaqed) [inline webui/src/lib/components/repositoryCreateForm.jsx:75]:
```suggestion
                    {!repoValid &&
```
(think I prefer this.)

**Review by arielshaqed — APPROVED** [review]
Neat, thanks!

(Please also wait for OSS TF approval)

**Comment by johnnyaug** [comment]
> Why are we adding the repository name to the prefix? to create a separate storage space for each repository (if so, why)?
@talSofer, yes.

Repositories should be in separate namespaces. We maybe not stress this enough in our docs, but some settings (like GC, branch protection) will override each other if repositories are in the same namespace.

File: webui/src/lib/components/repositoryCreateForm.jsx
Comment (johnnyaug) [reply webui/src/lib/components/repositoryCreateForm.jsx:75]: This is actually fixing a bug where the "Create repository" button becomes enabled even if you don't fill the storage namespace.

File: docs/reference/object-model.md
Comment (johnnyaug) [reply docs/reference/object-model.md:149]: Done

**Comment by talSofer** [comment]
> @talSofer, yes.
Repositories should be in separate namespaces. We maybe not stress this enough in our docs, but some settings (like GC, branch protection) will override each other if repositories are in the same namespace.

I see, can you open an issue to improve docs and maybe force this by code + migrate environments that are structured differently?

**Pull request merged**
